Move jailing code to core #1

IljaN opened the pull request:
<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please fill out below information carefully.

Please note that any kind of change first has to be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.
-->

## Description
- Read-only jailing of guest users is now provided by core
- Read-only groups can be configured in oc_appconfig under 'core' => 'read_only_groups' (coma seperated)

## Related Issue
#https://github.com/owncloud/guests/issues/1

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->


## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

butonic committed:
Update existing auth token on apache login

butonic committed:
regenerate session ID on apache login

butonic committed:
ocs needs to be able to login via apache

Commit:
Copy file to owner's trash when recipient moves out of share

Whenever a share recipient moves files or folders out of the share,
make a backup copy in the owner's trash just in case.

The metadata stays on the recipient's copy that was moved out.

DeepDiver1975 committed:
In some rare setups CSRF protection is done on the firewall level - e.g. F5 ASM

Commit:
Log files:scan exception, add InterruptedException

noveens committed:
New collumn of execution time added in db schema of oc_jobs

lordelix committed:
Add ignore_files to config,
test files against ignore_files list on upload

lordelix committed:
fix typo and indentation

noveens committed:
Finally completed adding a column to oc_jobs through migration.

noveens committed:
trailing space.

noveens committed:
Tests updation not required.

noveens committed:
New method added to store execution duration in oc_jobs

noveens committed:
wrong column fixed

imujjwal96 committed:
Allowing email for password recovery

noveens committed:
added quit option in commentstabview.js

noveens committed:
added quit option in notif in app.js

noveens committed:
added quit option in notif in file-upload.js

noveens committed:
added quit option in notif in fileinfomodel.js

noveens committed:
added quit option in notif in filelist.js

noveens committed:
added quit option in notif in filelist.js

noveens committed:
added quit option in notif in tagsplugin.js

noveens committed:
added quit option in notif in statusmanager.js

noveens committed:
added quit option in notif in external.js

noveens committed:
added quit option in notif in versionstabview.js

noveens committed:
added quit option in notif in notification.js

noveens committed:
changes according to the latest review.

Commit:
Register encryption storage wrapper in trashbin storage test

Because the encryption wrapper does some twisted stuff with methods like
moveFromStorage, we need to apply it here to make sure the behavior is
close to reality.

noveens committed:
called setExecutionTime at wrong place, fixed now.

noveens committed:
timeout removed since there is a button to close it

noveens committed:
default value of execution time set to -1, logging enabled.

noveens committed:
translation capability added

noveens committed:
Logging messages changed

noveens committed:
typo fixed

noveens committed:
logging messages format changed

noveens committed:
debug statements removed

imujjwal96 committed:
Remove redundancy following PR: #27168

noveens committed:
tests updated for oc_jobs

noveens committed:
defination of setExecutionTime() added in public.

noveens committed:
trying to pass jenkins

noveens committed:
trying to pass jenkins

noveens committed:
trying to pass jenkins

noveens committed:
test files updated

noveens committed:
small errors fixed

tomneedham committed:
Fix security warning to use correct admin ajax route

imujjwal96 committed:
Update to remove Test failure

Commit:
Keep sectionid in apps section in settings page

noveens committed:
config.sample.php permissions changed

settermjd committed:
A few minor changes to the CSRF protection documentation

Otherwise excellent!

davitol committed:
fix link to external storage

IljaN committed:
Move jailing code to core #1

tomneedham committed:
Fix the include for language codes

IljaN committed:
Fixing wrong indent #1

PVince81 commented:
Looks good so far.

- move the wrapper registration after the presetup hook firing
- use pipe separated group names (I thought we had this in sharing code already) because LDAP groups can have commas...
- only apply read only wrapper for logged in user ? note that setup FS might be called when mounting shares from other users in which case the wrapper might be superfluous (to be checked with a debugger)
- verify that fetching group memberships doesn't cause additional SQL queries. Normally the group manager already cached that info so we might be fine already

davitol committed:
link filesview global mounts to settings

lordelix committed:
Move blacklist declaration to lib/public/Files/FileInfo.php,
Rename *ignored to *blacklisted

davitol committed:
link filesview personal mounts to settings

IljaN commented:
@PVince81 I am not sure if I understand. The code is already applying the wrapper only for logged in users. 

> I thought we had this in sharing code already
Do you imply that there is some code I can reuse?

tomneedham committed:
Add icons to settings sections for UI (#27202)

* Add icons to settings sections for UI

* Less grey, more personal

tomneedham committed:
Fix call to get user lang in personal settings (#27197)

mmattel committed:
Defining App "cron" for "Invalidating tokens older than" message #27167 (#27201)

IljaN commented:
@PVince81 The Problem of this code is that when a external storage is Mounted for all users, and the guest user accesses it he suddenly gains write permission :-/

noveens committed:
Delete button now showing at lesser screen-widths than 768px

butonic committed:
allow configuring authType for DAV backend

noveens committed:
ensuring a square profile image cropper with initial width and height to be the minimum of width and height of parent image

PVince81 commented:
> The code is already applying the wrapper only for logged in users.

I meant "the currently logged in user as opposed to when `setupFS` is called for sharing users which aren't necessarily the current user". Need to test whether this actually happens.

PVince81 commented:
> Do you imply that there is some code I can reuse?

I remembered it wrong. Actually you need to use JSON instead.
Grep for the key "shareapi_exclude_groups_list" which match the sharing setting "Exclude groups from sharing" in the admin page.

>  The Problem of this code is that when a external storage is Mounted for all users, and the guest user accesses it he suddenly gains write permission

Good catch. I think the original code didn't care about such use cases (@butonic), but I think we should change it to also apply the wrapper on external storages. You might need to register the wrapper in a different location in the code, maybe close to the encryption or checksum wrapper as they also wrap every possible storage except sharing.

mmattel committed:
Update to use constants

IljaN commented:
Yeah the problem is that the SFTP stuff is done after setupFs is executed

lordelix committed:
Mocked blacklist_files for testing

lordelix committed:
Moved blacklist_files_regex into oc_config block

Commit:
Remove contact id in share with email code

The id is not used and would cause warnings when results come from the
system address book.

PVince81 commented:
The "oc_availability" storage also works for external storages, so I think you should register the jail one just after that one.

chagrawil committed:
Update TransferOwnership.php

date('c') uses colons which do not transfer to Windows clients.

mmattel committed:
fix typo in sftp public key

Commit:
Increase version due to changes in oc_jobs

tomneedham committed:
Fix profile test case dependancies in controller test

tomneedham committed:
Remove duplicate styles and scripts that are already imported, fixes #27229

mmattel committed:
Moving status code into a function

mmattel committed:
Printout status.php elements in Settings/General

Commit:
Fix indentation

settermjd committed:
Update the occ:encrypt-all prompt

The message isn't as clear, nor as succinct, as it could be.
Given that, this commit seeks to address both those points.
This commit was prompted by https://github.com/owncloud/documentation/pull/2835.

lordelix committed:
Add extensionpoint to sharedialogview
Write method overriding extension

mmattel committed:
PHPDoc fix

imujjwal96 committed:
Take email from user info instead of input

noveens committed:
HTTP 1.1 used when giving 403 and 404 errors while attempting to download non-existent file

Commit:
[tx-robot] updated from transifex

Commit:
Clear onpopstate handlers after unit tests

Fixes issue when running Karma tests in Firefox.

settermjd committed:
Updated UI messages relating to the encryption functionality

The reason for updating these messages, is that grammatically they're
not quite correct. They showed a combination of:

- Related phrases, which could be either separated or joined better
- Related sentences, but which should be expressed as separate ones

They were also missing full-stops to end the them.

Commit:
Fix public link download button

IljaN committed:
read_only_groups as json

lordelix committed:
Add a list of unknowns while inputting a search term

lordelix committed:
Rename OCA.Share extension point

PVince81 commented on lib/private/legacy/util.php:
```diff
@@ -208,9 +210,48 @@ public static function setupFS($user = '') {
+			'core',
+			'read_only_groups',
+			[]
+		));
```
`json_decode($something, true)`, else it will create a dummy class

PVince81 replied:
and I can imagine some people trying to use frontend side injection to make that class do wild things 😄

IljaN replied:
> and I can imagine some people trying to use frontend side injection to make that class do wild  >things :smile:

I will include some sanitizing at the write site?

PVince81 replied:
> I will include some sanitizing at the write site?

Yes, that too

davitol committed:
Add percentage to user's quota info (#27258)

* Add percentage to user's quota info

Fixes https://github.com/owncloud/core/issues/24011

* Do not show percentage if the quota is Unlimited

* translate quota

* correct condition and remove print_unescaped

Commit:
Fix syntax error in status manager

individual-it committed:
fix computerFileSize to parse correctly, prerapation to fix #25358 (#27257)

* fix computerFileSize to be mory picky

* more tests for computerFileSize

* codestyle fix

* fix double initialization

* pre-compile and store regexp in a hidden variable

as suggested by @PVince81 in Review https://github.com/owncloud/core/pull/27257#discussion_r103159658

* test strings with whitespaces

* trim string before parsing

VicDeo committed:
In the real world id is an initial filename

mmattel committed:
PHPDocfix for descriptiontext at getEditionString()

Commit:
[tx-robot] updated from transifex

IljaN committed:
#1 deserialize json as array

IljaN commented:
@PVince81 should @felixheidecke integrate the required changes for the frontend here or in a seperate PR? (Configuration for read_only_groups)

PVince81 commented:
@IljaN up to you guys

felixheidecke commented:
I prefer working on the same branch if the team consists of no more than two peeps :wink:

Commit:
[tx-robot] updated from transifex

individual-it committed:
fix for 25358 (#27275)

phisch committed:
Legacy example theme to app-theme (#27162)

* deleted example theme from themes directory

* added example theme app to .gitignore exceptions

* added info.xml to example theme app

* added example theme images, logos, defaults and styles

* fixed ownCloud name

* switched min version to 10

* bumped version to 1.0.0

* fixed oc min and max versions to 10.0

SergioBertolinSG committed:
Added functions to add files in local_storage

Commit:
Makefile improvements (#27222)

* Add make test-js-debug for continuous testing

Runs karma in continuous mode which makes it possible to debug in one
or multiple browsers

* Add make help command

IljaN commented:
@PVince81 This implementation prevents the creation of the user directory after first login. Any ideas?

IljaN commented:
Rebase Fail: continue here: https://github.com/owncloud/core/pull/27286
